whelk-core/src/main/groovy/whelk/converter/marc/NormalizeContentTypeStep.groovy

```diff
@@ -0,0 +1,27 @@
+package whelk.converter.marc
+
+class NormalizeContentTypeStep extends MarcFramePostProcStepBase {
+    def contentType = 'contentType'
+    def shape = [TYPE, contentType] as Set
+
+    void modify(Map record, Map thing) {
+        moveContentType(thing)
+    }
+
+    boolean moveContentType(Map thing) {
+        def work = thing.instanceOf
+        def moved = work?.hasPart?.removeAll { p ->
+            if (p.keySet() == shape) {
+                work[contentType] = ld.asList(work[contentType]) + p[contentType]
+                return true
+            }
+        }
+        if (work?.hasPart?.isEmpty()) {
+            work.remove('hasPart')
+        }
+        return moved
+    }
+
+    void unmodify(Map record, Map thing) {
```

Comment:
_If_ we need to have the same repeated 336:s on revert, we can add back repeated contentType values as parts here.

Reply:
See updated description.